source/braille.py

```diff
@@ -2452,6 +2501,42 @@ def _set_displaySize(self, value):
 			f"Can't set displaySize to {value}, consider registering a handler to filter_displaySize",
 		)
 
+	displayNumRows: int
+	_cache_displayNumRows = True
+
+	def _get_displayNumRows(self) -> int:
+		"""
+		Returns the number of rows on the display.
+		Handlers can register themselves to L{filter_displayNumRows} to change this value on the fly.
+		Therefore, this is a read only property and can't be set.
+		"""
+		numRows = self.display.numRows if self.display else 0
+		currentDisplayNumRows = filter_displayNumRows.apply(numRows)
+		if self._displayNumRows != currentDisplayNumRows:
+			displaySizeChanged.notify(displaySize=self._displaySize, displayNumRows=currentDisplayNumRows)
+			self._displayNumRows = currentDisplayNumRows
+		return currentDisplayNumRows
+
+	def _set_displayNumRows(self, value: int):
+		raise AttributeError(
+			f"Can't set displayNumRows to {value}, consider registering a handler to filter displayNumRows",
+		)
+
+	displayNumCols: int
+	_cache_displayNumCols = True
+
+	def _get_displayNumCols(self) -> int:
+		"""
+		Returns the number of columns on the display.
+		This is calculated from displaySize and displayNumRows.
+		"""
+		return self.displaySize // self.displayNumRows
```

Comment:
Ugh, I'm afraid there is an edge case here. Imagine the case where someone registers a handler for `filter_displayNumRows` without lowering the display size. That could potentially cause the handler to think that there is only one row with 320 cells, for example.
Thinking about that, it should probably be investigated how the braille viewer behaves with all this.

Reply:
What if displayNumRows and displayNumCols began as the display's numRows and numCols respectively, and then were filtered on a new filter_displayDimensions extension point which takes a tuple of numRows and numCols. 
Then displaySize is calculated from displayNumRows and displayNumCols. It is run through the original filter_displaySize extension point. If it is found to change, then we use that instead, and force displayNumRows to 1 and displayNumCols to the filtered displaySize. 
So for older copies of NVDA / remote that only support filter_displaySize, you get one row of displaySize. For modern NVDA / remote you'll get proper rows and columns.
The recommendation would be newer NVDA / remote only registers displayDimentions and not displaySize, though if they did make it be numRows * numCols that would also still be okay.

Reply:
I think this makes sense, though for numRows when filtering with display size, you could also still consider `min(self.display.numRows, math.ceil(self.displaySize / self.display.numCols))`. Or do you think this has issues?